temporal-sdk/src/main/java/io/temporal/common/Priority.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.common;
+
+import java.util.Objects;
+
+/**
+ * Priority contains metadata that controls the relative ordering of task processing when tasks are
+ * backed up in a queue. The affected queues depend on the server version.
+ *
+ * <p>Priority is attached to workflows and activities. By default, activities and child workflows
+ * inherit Priority from the workflow that created them, but may override fields when an activity is
+ * started or modified.
+ *
+ * <p>For all fields, the field not present or equal to zero/empty string means to inherit the value
+ * from the calling workflow, or if there is no calling workflow, then use the default value.
+ */
+@Experimental
+public class Priority {
```

Comment:
Not a fan of one parameter builders, but since we know this field will expend we should make this a builder.

Reply:
Agreed, this object should be immutable. This likely includes making the constructor private, but we can have a static helper that creates it from just the integer.

Reply:
Converted this to a builder to be consistent with the rest of the SDK